### PR TITLE
support writing config to non-default.yml config files

### PR DESCRIPTION
**Pull request by unkempthenry**
Changes: 

* previously changes to in memory config would always be written to default.yml even if caldera was started with a different conf file
* only write changes if the config has actually changed (this prevents re-ordering the conf file unecessarily) 

Thoughts: 

Not sure I like re-writing the config file that much, since it reorders the option and will eat any comments in it.  Maybe we could instead write a `<conf-name>-overide.yml` file and then our config loading option would get the config from the original file, and then just `.update()` it with the overrides.

### CODE REVIEW

File: app/service/app_svc.py [hunk 1]
```diff
@@ -143,8 +144,11 @@ def __init__(self, application):
     """ PRIVATE """
 
     async def _save_configuration(self):
-        with open('conf/default.yml', 'w') as config:
-            config.write(yaml.dump(self.get_config()))
+        with open('conf/{}.yml'.format(self.config_name), 'w') as config:
+            old_config = yaml.load(config, Loader=yaml.FullLoader)
+        if self.get_config() != old_config:
+            with open('conf/default.yml', 'w') as config:
```
Comment (wbooth) [inline app/service/app_svc.py]: should this use self.config_name ?

Comment (privateducky) [inline app/service/app_svc.py]: should we update default.yml in this case - or the config name that we started the app with? seems like the latter..

Comment (unkempthenry) [reply app/service/app_svc.py]: 100% right; fixed now.

Comment (unkempthenry) [reply app/service/app_svc.py]: 100% right; fixed now.

**Comment by privateducky** [comment]
closing this. per an internal conversation, if we make a future change to config management, i'd rather us store them in the data_svc:ram object and load from there upon reboots. this would involve changing default.yml to init.yml and only "inducing" it on the first server start.

**Pull request closed without merging**